### DISCUSSION TITLE
How to preserve locked segments present in my sdlxliff files while converting to xliff

### DISCUSSION THREAD

**Discussion by Abhishek9934** [maxprograms-com/OpenXLIFF#35] [General]
Hi, I am working on a translation based project in which I want to skip translations for locked segments present in the xliff, sldxliff , mqxliff files. I am using the openXliff convert to generate xlf  and skl  files on which further processing is done. 
But when I use convert, the locked="true" attributes are not getting preserved in the generated xlf files, is there a way to preserve the locked attributes while conversion ?

-Thanks

**Comment by rmraya** [comment]
Upload sample files that I can use for reproducing.

**Comment by Abhishek9934** [comment]
Sample files links - 

https://drive.google.com/file/d/1IquCaLfFjMCGLOAWdnhYC7sb0XXjICmE/view?usp=drive_link

https://drive.google.com/file/d/17FMiVN5orAvCJVuu-Os-ndz38FJANc0V/view?usp=drive_link

**Comment by rmraya** [comment]
Are you talking about the segments that contain fragments like this:
```
  <sdl:seg-defs>
    <sdl:seg id="73" locked="true" conf="Translated" origin="tm"
        origin-system="CN-3000-6000_EN-LV_240308" percent="100"
        struct-match="true">
```

**Reply by Abhishek9934** [reply]
Yes, for some trans-units segments present in the file, it has these locked = true properties present in the fragments.

**Comment by rmraya** [comment]
OK, but are you aware that there is no `"locked"` attribute in the XLIFF standard? 

All I can do is set `state="signed-off"` for locked segments and maybe add `state-qualifier="leveraged-tm"` where `origin="tm"`

**Reply by Abhishek9934** [reply]
Having the state="signed-off" would help.

**Comment by rmraya** [comment]
Changes implemented in release 3.25.0: 

- If you generate XLIFF 1.2, `ts` attribute of `<trans-unit>` is set to `locked`.
- if you generate XLIFF 2.x, `subState` attribute of `<segment>` is set to `openxliff:locked`

In both cases `state` attribute has a value compatible with the one defined in the SDLXLIFF file.

Let me know if this works.
